library/src/main/java/com/datatorrent/lib/filter/FilterOperator.java

```diff
@@ -0,0 +1,177 @@
+
+/**
+ * <b>FilterOperator</b>
+ * Filter Operator filter out tuples based on defined condition
+ *
+ * <b>Parameters</b>
+ * - condition: condition based on expression langugage
+ *
+ * <b>Input Port</b> takes POJOs as an input
+ *
+ * <b>Output Ports</b>
+ * - truePort emits POJOs meeting the given condition
+ * - falsePort emits POJOs not meeting the given condition
+ * - error port emits any error situation while evaluating expression
+ * 
+ */
+@InterfaceStability.Evolving
+public class FilterOperator extends BaseOperator implements Operator.ActivationListener
+{
+  protected String condition;
+
+  protected Class<?> inClazz = null;
+
+  private Expression<Boolean> expr = null;
+
+  @AutoMetric
+  protected long trueTuples;
+
+  @AutoMetric
+  protected long falseTuples;
+
+  @AutoMetric
+  protected long errorTuples;
+
+  @InputPortFieldAnnotation(schemaRequired = true)
+  public transient DefaultInputPort<Object> input = new DefaultInputPort<Object>()
+  {
+    public void setup(PortContext context)
+    {
+      inClazz = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+
+    @Override
+    public void process(Object t)
+    {
+      processTuple(t);
+    }
+  };
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> truePort = new DefaultOutputPort<Object>();
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> falsePort = new DefaultOutputPort<Object>();
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> error = new DefaultOutputPort<Object>();
+
+  @Override
+  public void activate(Context context)
+  {
+    createExpression();
+  }
+
+  @Override
+  public void deactivate()
+  {
+    expr = null;
+  }
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    errorTuples = trueTuples = falseTuples = 0;
+  }
+
+  /**
+   * createExpression: create an expression from condition of POJO fields
+   * Override this function for custom field expressions
+   */
+  protected void createExpression()
+  {
+    logger.info("Creating an expression for condition {}", condition);
+    expr = PojoUtils.createExpression(inClazz, condition, Boolean.class);
+  }
+
+  /**
+   * evalExpression: Evaluate condition/expression
+   * Override this function for custom condition evaluation
+   */
+  protected Boolean evalExpression(Object t)
+  {
+    return expr.execute(t);
+  }
+
+  /**
+   * handleFilter: emit POJO meeting condition on truePort
+   * and if it did not meet condition then on falsePort
+   */
+  private void processTuple(Object t)
+  {
+    try {
+      if (evalExpression(t)) {
+        truePort.emit(t);
+        trueTuples++;
+      } else {
+        falsePort.emit(t);
+        falseTuples++;
+      }
+    } catch (Exception ex) {
+      logger.error("Error in expression eval: {}", ex.getMessage());
+      logger.debug("Exception: ", ex);
```

Comment:
Basically the idea is to limit extended error messages in data path. This exception is on incorrect tuples, hence stacktrace would probably fill logs quickly if we start hitting the issue and make it hard to analyze debug logs. Hence we have added single line message in error and if required entire stacktrace can be retrieved in debug mode.